src/NUnitFramework/framework/Attributes/ValuesAttribute.cs

```diff
@@ -101,7 +103,51 @@ public ValuesAttribute(params object[] args)
         /// <param name="parameter">The parameter of a parameterized test.</param>
         public IEnumerable GetData(Type fixtureType, ParameterInfo parameter)
         {
+            if(data.Length == 0)           
+                return GenerateData(parameter.ParameterType);
+            else           
             return ParamAttributeTypeConversions.ConvertData(data, parameter.ParameterType);
         }
+
+        /// <summary>
+        /// To generate data for Values attribute, in case no data is provided.
+        /// </summary>
+        /// <param name="targetType"></param>
+        /// <returns></returns>
+        private static IEnumerable GenerateData(Type targetType)
```

Comment:
Personally, I think that the `param` and `returns` are unnecessary if they don't contain any information.